akka-stream-tests/src/test/scala/akka/stream/scaladsl/FlowCollectSpec.scala

```diff
@@ -23,6 +31,20 @@ class FlowCollectSpec extends AkkaSpec with ScriptedTest {
       TestConfig.RandomTestRange foreach (_ ⇒ runScript(script, settings)(_.collect { case x if x % 2 == 0 ⇒ (x * x).toString }))
     }
 
+    "restart when Collect throws" in {
+      val ex = new Exception("test") with NoStackTrace
```

Comment:
There is a test exception available in the testkit already, `akka.stream.testkit.Utils.TE`